zapier tweaks

snopoke opened the pull request:
Just some things I noticed while working on something else

snopoke committed:
use util that does domain check

This was missing a domain check

snopoke committed:
use HttpResponseBadRequest

snopoke committed:
change 403 to 400

403 seems like the wrong response code for these errors

snopoke commented on corehq/apps/zapier/views.py:
```diff
@@ -118,7 +117,7 @@ def post(self, request, *args, **kwargs):
 
         if not case_type or not owner_id or not domain or not case_name:
-            return HttpResponseForbidden('Please fill in all required fields')
+            return HttpResponseBadRequest('Please fill in all required fields')
```
changes from 403 -> 400

snopoke commented on corehq/apps/zapier/views.py:
```diff
@@ -160,13 +159,10 @@ def post(self, request, *args, **kwargs):
-            case = CaseAccessors(domain).get_case(case_id)
-        except CaseNotFound:
-            return HttpResponseForbidden("Could not find case in domain")
+        case = get_case_or_404(domain, case_id)
```
changes from 403 -> 404

snopoke commented on corehq/apps/zapier/views.py:
```diff
@@ -160,13 +159,10 @@ def post(self, request, *args, **kwargs):
 
         if not case.type == case_type:
-            return HttpResponseForbidden("Case type mismatch")
+            return HttpResponseBadRequest("Case type mismatch")
```
changes from 403 -> 400

pr33thi commented on corehq/util/view_utils.py:
```diff
@@ -129,3 +133,23 @@ def reverse(viewname, params=None, absolute=False, **kwargs):
+def get_form_or_404(domain, id):
+    try:
+        form = FormAccessors(domain).get_form(id)
+        if form.domain != domain:
```
should this have a similar `form.is_deleted` check?

snopoke replied:
Seems reasonable: e532ad7

snopoke committed:
resolve import cycle

snopoke committed:
check if form is deleted

pr33thi reviewed (approved):
Travis failure looks related

snopoke committed:
update status codes in tests

snopoke committed:
fix access test

This test was actually broken but since both errors returned the
same code it was still passing
